src/Parsers/IAST.h

```diff
@@ -195,12 +195,12 @@ class IAST : public std::enable_shared_from_this<IAST>, public TypePromotion<IAS
         bool one_line;
         bool always_quote_identifiers = false;
         IdentifierQuotingStyle identifier_quoting_style = IdentifierQuotingStyle::Backticks;
-        bool show_secrets = true; /// Show secret parts of the AST (e.g. passwords, encryption keys).
+        bool show_secrets; /// Show secret parts of the AST (e.g. passwords, encryption keys).
 
         // Newline or whitespace.
         char nl_or_ws;
 
-        FormatSettings(WriteBuffer & ostr_, bool one_line_, bool show_secrets_ = true)
+        FormatSettings(WriteBuffer & ostr_, bool one_line_, bool show_secrets_ = false)
```

Comment:
Let's separate things and not do everything in the same PR. Adding a CLI flag can be the goal of the first PR, and making `show_secrets = false` the default in `FormatSettings` can be the goal of the second PR, and then if you want also to improve that hiding in some way and change `ASTAuthenticationData` or `ASTFunction` it will be the third PR.